stock_measuring_device_zippcube/controllers/main.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+import os
+
+from werkzeug import exceptions
+
+from odoo import _, http
+from odoo.exceptions import AccessError, MissingError, UserError
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class ZippcubeController(http.Controller):
+    DEVICE_SECRET_KEY = "ZIPPCUBE_SECRET"
+
+    weight_keys = ("weight",)
+    measures_keys = ("length", "width", "height")
+    expected_keys = ("secret", "barcode") + weight_keys + measures_keys
+
+    @http.route(
+        "/stock/zippcube/<string:zippcube_device_name>/measurement",
+        type="json",
+        auth="none",
+    )
+    def measurement(self, zippcube_device_name):
+        data = request.jsonrequest
+        _logger.info("/measurement, data received: {}".format(data))
+
+        env = request.env(su=True)
+        device = env["measuring.device"].search(
+            [("name", "=", zippcube_device_name), ("device_type", "=", "zippcube")],
+            limit=1,
+        )
+        if not device:
+            raise MissingError(
+                _("No such Zippcube with name {}.").format(zippcube_device_name)
+            )
+
+        keys_missing = set(self.expected_keys) - set(data)
+        if keys_missing:
+            error_msg = _("Wrong data format: {}. Keys missing: {}.").format(
+                data, keys_missing
+            )
+            _logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        self._check_secret(data["secret"])
+        # convert the float values passed as strings to floats
+        data = self._convert_floats(data)
+        with device.work_on(device._name) as work:
```

Comment:
the device model should have a method `_update_packaging_measures` which forwards the call.
In any case here you could use `_get_measuring_device` :wink: